head/src/main/kotlin/io/qalipsis/core/head/campaign/AbstractCampaignExecutor.kt

```diff
@@ -167,7 +169,7 @@ internal abstract class AbstractCampaignExecutor<C : CampaignExecutionContext>(
         val campaignStartState = create(runningCampaign, factories, scenarios)
         log.info { "Starting the campaign ${configuration.name} with scenarios ${scenarios.map { it.name }} on factories ${factories.map { it.nodeId }}" }
         campaignStartState.inject(campaignExecutionContext)
-        processingMutex.withLock {
+        lockProvider.withLock(campaignKey = runningCampaign.key) {
```

Comment:
Since the campaign is not really started yet, there is no need to do it in a lock.

Reply:
@ericjesse taking this out has an effect on the 2secs atMostFor of the redisLokk. For some reason, if taken out the 2Seconds becomes a really high value atMostFor to attain and the application throws a timeout exception